plugins/inputs/opcua_listener/subscribe_client.go

```diff
@@ -32,6 +33,54 @@ type SubscribeClient struct {
 	processingCancel context.CancelFunc
 }
 
+func AssignConfigValuesToRequest(req *ua.MonitoredItemCreateRequest, monParams *input.MonitoringParameters, log telegraf.Logger) {
+	req.RequestedParameters.SamplingInterval = float64(time.Duration(monParams.SamplingInterval) / time.Millisecond)
+
+	if monParams.QueueSize != "" {
+		queueSize, err := strconv.ParseUint(monParams.QueueSize, 10, 32)
+		if err == nil {
+			req.RequestedParameters.QueueSize = uint32(queueSize)
+		} else {
+			log.Debugf(
+				"Could not parse queue_size %s into uint32. Standard value %d used instead.",
+				monParams.QueueSize,
+				req.RequestedParameters.QueueSize)
+		}
+	}
+
+	if monParams.DiscardOldest != "" {
+		discardOldest, err := strconv.ParseBool(monParams.DiscardOldest)
+		if err == nil {
+			req.RequestedParameters.DiscardOldest = discardOldest
+		} else {
+			log.Debugf(
+				"Could not parse discard_oldest %s into bool. Standard value %t used instead.",
+				monParams.DiscardOldest,
+				req.RequestedParameters.DiscardOldest)
+		}
+	}
+
+	if monParams.DataChangeFilter.Trigger != "" {
+		trigger := ua.DataChangeTriggerFromString(monParams.DataChangeFilter.Trigger)
+		deadbandType := uint32(ua.DeadbandTypeFromString(monParams.DataChangeFilter.DeadbandType))
+		deadbandValue, err := strconv.ParseFloat(monParams.DataChangeFilter.DeadbandValue, 64)
+
+		if err == nil && deadbandValue != 0 {
```

Comment:
so only if the deadband value is not 0 do we set this, however you show deadband values as 0 in one of the examples above. You should document this limitation as well.

Reply:
I suggest changing the values in the examples to 0 (again to avoid breaking anything), changing the condition to `... deadbandValue > 0` and adding this to the documentation in 'Data change filter' -> 'deadband_value'.